File: caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    if(argc!=2)
    {
        printf("error\n");
        return 1;
    }
    int i;
    printf("plaintext: ");
    string s = get_string();
    string k = argv[1];
    int n = atoi(k);
    n = n%26;
    
    for(i = 0;i < strlen(s);i++)
    {
        if(isalpha(s[i]))
        {
            if(isupper(s[i]))
            {
                s[i] = ((((s[i] -'A') + n) % 26) + 'A');
            }
                
            if(islower(s[i]))
            {
                s[i] = ((((s[i] -'a') + n) % 26) + 'a');
            }
        }
    }

    printf("ciphertext: %s\n",s);
    return 0;
}

File: caesar.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("Error!")
        exit(1)
    k = int(sys.argv[1])
    t = []
    print("plaintext: ",end = "")
    s = cs50.get_string()
    k = k%26
    for c in s:
        if(c.isalpha):
            if(c.isupper()):
                a = chr(((ord(c) - 65 + k) % 26) + 65)
                t.append(a)
            elif(c.islower()):
                a = chr(((ord(c) - 97 + k) % 26) + 97)
                t.append(a)
            else:
                t.append(c)
    print("ciphertext: ",end = "")
    print("".join(t))

if __name__ == "__main__":
    main()
            
            
